snippets/hashNode.md

```diff
@@ -0,0 +1,14 @@
+### hashNode
+
+Creates a hash for a value using the [SHA-256](https://en.wikipedia.org/wiki/SHA-2) algorithm.
+
+Use `crypto` API to create a hash for the given value.
+
+```js
+const crypto = require('crypto');
+const hashNode = val => crypto.createHash('sha256').update(val).digest('hex');
```

Comment:
# Note
We could make this async and return a promise
```js
const hashNode = val =>
  new Promise( resolve => setTimeout(() => resolve(crypto.createHash('sha256').update(val).digest('hex')),0))
```

Reply:
I'd rather make the browser one non-async (which is not possible), as this is a lot easier to use than the promise-based version. Usually hashing won't take extremely long and, in most cases I have needed it, it's something you do not want to have running on the background but rather have done immediately. Might be wrong, though.

Reply:
This make the function have the same footprint as the other hashing algorithm so  that you could actively replace one with the other without any other changes to your code. However if we don't need it async, at least resolving it in a promise gives it that capability. I just wanted to replicate the browser behavior as much as possible so that when someone uses it in an async environment( such as a webserver or a request heavy script with multiple instances per something ) they wouldn't hold up other parts of the script while it runs the SHA and Buffer parts of the code unless the eventloop was empty.

NOTE: secure hashing and most crypto functions are actually quite slow( comparatively ). It's why I'm a fan of using the built in apis to make them operate later. If you're using it for a one time use script then it's perfectly fine, however in a webserver you'll either want a callback, or a promise so that it can run when other things are done because of how awfully slow it can be.

Reply:
Convincing argument, I'll swap it out to an async version for consistency and the rest of the reasons outlined above.

Reply:
> However if we don't need it async, at least resolving it in a promise gives it that capability. I just wanted to replicate the browser behavior as much as possible so that when someone uses it in an async environment( such as a webserver or a request heavy script with multiple instances per something ) they wouldn't hold up other parts of the script while it runs the SHA and Buffer parts of the code unless the eventloop was empty.

To be honest @skatcat31 but at that point it really depends on how the user is using this current snippet. Hopefully the user would be playing around with said snippet and using it for educational purposes and not trying to use this exact code in a production/server environment. Because there is soo many way's these can go wrong depending on the use case

Reply:
@kingdavidmartins fair point! At that point it should be
> However if we don't need it async, at least resolving it in a promise gives it that capability.

and I should have stopped with that argument. Hopefully at least having similar functions that behave similarly helps the learning process along

Reply:
That being said security/crypto in general is never really handled by a beginner. So I don't think there's really anything to worry about

Reply:
> Hopefully at least having similar functions that behave similarly helps the learning process along

@skatcat31 Yup Yup Agreed!!